src/v/compaction/utils.cc

```diff
@@ -61,7 +61,7 @@ model::record_batch make_placeholder_batch(model::record_batch_header& hdr) {
     new_hdr.first_timestamp = hdr.first_timestamp;
```

Comment:
> model: move reset_size_checksum_metadata() to member function of …
> …`record_batch_header`

please be mindful of the commit message length. pro-tip: vim will highlight the overflow